searx/engines/wikipedia.py

```diff
@@ -41,51 +31,37 @@ def url_lang(lang):
 # do search-request
 def request(query, params):
     if query.islower():
-        query = u'{0}|{1}'.format(query.decode('utf-8'), query.decode('utf-8').title()).encode('utf-8')
+        query = query.title()
 
-    params['url'] = search_url.format(query=urlencode({'titles': query}),
+    params['url'] = search_url.format(title=quote(query),
                                       language=url_lang(params['language']))
 
+    params['headers']['User-Agent'] = searx_useragent()
+
     return params
 
 
 # get response from search-request
 def response(resp):
-    results = []
-
-    search_result = loads(resp.text)
-
-    # wikipedia article's unique id
-    # first valid id is assumed to be the requested article
-    if 'pages' not in search_result['query']:
-        return results
-
-    for article_id in search_result['query']['pages']:
-        page = search_result['query']['pages'][article_id]
-        if int(article_id) > 0:
-            break
-
-    if int(article_id) < 0 or 'disambiguation' in page.get('pageprops', {}):
+    if not resp.ok:
         return []
 
-    title = page.get('title')
-
-    image = page.get('thumbnail')
-    if image:
-        image = image.get('source')
+    results = []
+    api_result = loads(resp.text)
 
-    summary = page.get('extract', '').split('\n')[0].replace('()', '')
+    # skip disambiguation pages
+    if api_result['type'] != 'standard':
```

Comment:
The type ```disambiguation``` is ignored. I don't know if this is intent ?

Example with ```!wp :en search```

```curl -L -vvv -X GET "https://en.wikipedia.org/api/rest_v1/page/summary/search" -H  "accept: application/json; charset=utf-8; profile="https://www.mediawiki.org/wiki/Specs/Summary/1.4.2""```

Reply:
Yeah. Otherwise, with your example you would end up with an infobox that just says: 
> Searching or search may refer to:  

...and that's it, but that would be rather useless in my opinion. Although maybe I should still add the URL to the normal results in that case...